code/modules/surgery/surgery_helpers.dm

```diff
@@ -83,27 +83,40 @@
 
 /proc/attempt_cancel_surgery(datum/surgery/S, obj/item/I, mob/living/M, mob/user)
 	var/selected_zone = user.zone_selected
+
 	if(S.status == 1)
 		M.surgeries -= S
 		user.visible_message("<span class='notice'>[user] removes [I] from [M]'s [parse_zone(selected_zone)].</span>", \
 			"<span class='notice'>You remove [I] from [M]'s [parse_zone(selected_zone)].</span>")
 		qdel(S)
-	else if(S.can_cancel)
+		return
+
+	if(S.can_cancel)
 		var/required_tool_type = TOOL_CAUTERY
 		var/obj/item/close_tool = user.get_inactive_held_item()
 		var/is_robotic = S.requires_bodypart_type == BODYPART_ROBOTIC
+
 		if(is_robotic)
 			required_tool_type = TOOL_SCREWDRIVER
-		if(close_tool && close_tool.tool_behaviour == required_tool_type || iscyborg(user))
-			if (ishuman(M))
-				var/mob/living/carbon/human/H = M
-				H.bleed_rate = max( (H.bleed_rate - 3), 0)
-			M.surgeries -= S
-			user.visible_message("<span class='notice'>[user] closes [M]'s [parse_zone(selected_zone)] with [close_tool] and removes [I].</span>", \
-				"<span class='notice'>You close [M]'s [parse_zone(selected_zone)] with [close_tool] and remove [I].</span>")
-			qdel(S)
-		else
+
+		if(iscyborg(user))
+			close_tool = locate(/obj/item/cautery) in user.held_items
+			if(!close_tool)
+				to_chat(user, "<span class='warning'>You need to equip a cautery in an inactive slot to stop [M]'s surgery!</span>")
+				return
+
+		if((!close_tool || close_tool.tool_behaviour != required_tool_type) && !iscyborg(user)) // don't check tool type on cyborgs

```

Comment:
this can be an else / else if, you shouldn't need to ensure they're not a cyborg if you've ensured they are a cyborg above

Reply:
Done.